### PR TITLE
corechecks: Fix a crash in PreCallValidateCmdBindDescriptorSets

### LINKED ISSUE #3014: crash on descriptor binding
Binding a descriptor set containing dynamic buffer descriptors where the buffer is null but the dynamic offset is nonzero causes a crash.

**Environment:**
 - OS: Linux
 - GPU: Any
 - SDK or header version if building from repo:  f4fb2a0b00d10942997e3ff7b93c40b1409a8b2b
 - Options enabled (synchronization, best practices, etc.): Default

This is a bit of a weird one according to spec since it doesn't seem to be explicitly covered, but based on my reading, it appears the validation for the dynamic offset should occur after determining whether the descriptor is null instead of only considering whether the descriptor may be null if the dynamic offset is zero.

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
The checking for dynamic offsets was crashing if the descriptor's buffer was VK_NULL_HANDLE. Fixes #3014

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
@sfricke-samsung, this is just a minimal fix of the crash. The test case shows that we handle VK_NULL_HANDLE buffers elsewhere.  I can't find anything in the spec to suggest doing something else. Based on your prior work here, what do you think?

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 1]
```diff
@@ -5161,6 +5161,100 @@ TEST_F(VkLayerTest, InvalidDynamicDescriptorSet) {
     m_commandBuffer->end();
 }
 
+TEST_F(VkLayerTest, DynamicOffsetWithNullBuffer) {
+    // Create a descriptorSet w/ dynamic descriptors where 1 binding is inactive, but all have null buffers
```
Comment (ncesario-lunarg) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:5165]: `TEST_DESCRIPTION`? Though, I don't know if I've ever seen the `TEST_DESCRIPTION` text outside of the C++ source...

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp:5165]: looks like you get it if you use --gtest_output=json:foo.json

```
{
  "tests": 1,
  "failures": 0,
  "disabled": 0,
  "errors": 0,
  "timestamp": "2021-07-22T16:43:31Z",
  "time": "1.537s",
  "name": "AllTests",
  "testsuites": [
    {
      "name": "VkLayerTest",
      "tests": 1,
      "failures": 0,
      "disabled": 0,
      "errors": 0,
      "timestamp": "2021-07-22T16:43:31Z",
      "time": "1.531s",
      "testsuite": [
        {
          "name": "PSOPolygonModeInvalid",
          "status": "RUN",
          "result": "COMPLETED",
          "timestamp": "2021-07-22T16:43:31Z",
          "time": "1.531s",
          "classname": "VkLayerTest",
          "description": "Attempt to use invalid polygon fill modes."
        }
      ]
    }
  ]

```

Comment (ncesario-lunarg) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp]: Good to know :)

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**